learning/dashboard/WHAT IS JAVASCRIPT/Basics of JavaScript.md

```diff
@@ -0,0 +1,941 @@
+selecting nodes in the DOM tree f rom other nodes. You're probably
+already familiar with several methods for accessing elements in the DO M
+tree by their id, class, or tag name. You can use methods like
+document.querySelector() and document .getElementById() to do so.
+
+There are other methods you can use in conjunction, to navigate the DOM
+in more efficient and robust wa ys. As you can imagine, it's better to
+search from an already-known point on a map than do a full search.
+
+For example, selecting a child element from its parent is easier and
+more efficient than searching for it thr oughout the entire tree.
+
+A Sample Document to Traverse\
+Once you have access to a given node in the DOM tree, you can access its
+related nodes in different way s. You can move downwards, upwards, or
+sideways in the DOM tree from your selected node.
+
+The first method involves searching for an element starting with a
+top-most node (like the document node) and moving downwards.
+
+The second way is the opposite: you move from an inner element up the
+tree, searching for an outer elem ent. The last method is when you
+search for an element from another element at the same level (meaning
+the two elements are siblings) in the document tree.
+
+### **To demonstrate, consider this example HTML document:**
+``` html
+<!DOCTYPE html\>
+<html lang=\"en\"\>
+
+<head\>
+<meta charset=\"UTF-8\" /\>
+<meta http-equiv=\"X-UA-Compatible\" content=\"IE=edge\" /\>
+<meta name=\"viewport\" content=\"width=device-width,
+initial-scale=1.0\" /\> \<title\>Sample page\</title\>
+</head\>
+
+<body\>
+<main\>
+<h1\>My Page Title\</h1\>
+<p\>Nice caption goes here\</p\>
+
+<article class=\"first\_\_article\"\>
+<h2\>List of amazing fruits\</h2\>
+<p\>Must eat fruits\</p\>
+<div class=\"wrapper-1\"\>
+<ul class=\"apple-list\"\>
+<li class=\"apple\"\>Apples\</li\>
+<li class=\"orange\"\>Oranges\</li\>
+<li class=\"avocado\"\>Avocados\</li\>
+<li class=\"grape\"\>
+
+ Grapes
+
+<ul\>
+<li class=\"type-1\"\>Moon drops\</li\>
+<li\>Sultana\</li\>
+<li\>Concord\</li\>
+<li\>Crimson Seedless\</li\>
+</ul\>
+</li\>
+<li class=\"banana\"\>Bananas\</li\>
+</ul\>
+<button class=\"btn-1\"\>Read full list\</button\>
+</div\>
+</article\>
+
+<article class=\"second\_\_article\"\>
+<h2\>Amazing places in Kenya\</h2\>
+<p\>Must visit places in Kenya\</p\>
+
+<div class=\"wrapper-2\"\>
+<ul class=\"places-list\"\>
+<li\>Maasai Mara\</li\>
+<li\>Diani Beach\</li\>
+<li\>Watamu Beach\</li\>
+<li\>Amboseli national park\</li\>
+<li\>Lake Nakuru\</li\>
+</ul\>
+
+<button class=\"btn-2\"\>Read full list\</button\> \</div\>
+</article\>
+</main\>
+</body\>
+
+</html\>
+```
+
+### Traversing the DOM Downwards
+You can traverse the DOM downwards using one of two methods. The first
+one is the common selector m ethod (element.querySelector or
+element.querySelectorAll). Secondly, you can use the children or childN
+odes property. There are also two other special properties, namely,
```

Comment:
Check the formatting. `childN odes` and many such issues in the document. Looks like the content didn't convert properly.